crates/sui-sdk/examples/utils.rs

```diff
@@ -0,0 +1,305 @@
+
+    // now we programatically build the transaction through several commands
+    let mut ptb = ProgrammableTransactionBuilder::new();
+    // first, we want to split the coin, and we specify how much SUI (in MIST) we want
+    // for the new coin
+    let split_coin_amount = ptb.pure(1000u64)?; // note that we need to specify the u64 type here
+    ptb.command(Command::SplitCoins(
+        Argument::GasCoin,
+        vec![split_coin_amount],
+    ));
+    // now we want to merge the coins (so that we don't have many coins with very small values)
+    // observe here that we pass Argument::Result(0), which instructs the PTB to get
+    // the result from the previous command
+    ptb.command(Command::MergeCoins(
+        Argument::GasCoin,
+        vec![Argument::Result(0)],
+    ));
+
+    // we finished constructing our PTB and we need to call finish
+    let builder = ptb.finish();
+
+    // using the PTB that we just constructed, create the transaction data
+    // that we will submit to the network
+    let tx_data = TransactionData::new_programmable(
+        *sender,
+        vec![coin.object_ref()],
+        builder,
+        max_gas_budget,
+        gas_price,
+    );
+
+    // sign & execute the transaction
+    let keystore = FileBasedKeystore::new(&sui_config_dir()?.join(SUI_KEYSTORE_FILENAME))?;
+    let signature = keystore.sign_secure(sender, &tx_data, Intent::sui_transaction())?;
+
+    let transaction_response = sui
+        .quorum_driver_api()
+        .execute_transaction_block(
+            Transaction::from_data(tx_data, Intent::sui_transaction(), vec![signature]),
+            SuiTransactionBlockResponseOptions::new(),
+            Some(ExecuteTransactionRequestType::WaitForLocalExecution),
+        )
+        .await?;
+    Ok(transaction_response.digest)
+}
+
+pub async fn retrieve_wallet() -> Result<WalletContext, anyhow::Error> {
+    let wallet_conf = sui_config_dir()?.join(SUI_CLIENT_CONFIG);
+    let keystore_path = sui_config_dir()?.join(SUI_KEYSTORE_FILENAME);
+
+    // check if a wallet exists and if not, create a wallet and a sui client config
+
+    if !keystore_path.exists() {
+        let keystore = FileBasedKeystore::new(&keystore_path)?;
+        keystore.save()?;
+    }
+
+    if !wallet_conf.exists() {
+        let keystore = FileBasedKeystore::new(&keystore_path)?;
+        let mut client_config = SuiClientConfig::new(keystore.into());
+        client_config.add_env(SuiEnv {
+            alias: "testnet".to_string(),
+            rpc: "https://fullnode.testnet.sui.io:443".to_string(),
+            ws: None,
+        });
+
+        client_config.add_env(SuiEnv::devnet());
+
+        client_config.add_env(SuiEnv {
+            alias: "localnet".to_string(),
+            rpc: "http://127.0.0.1:9000".to_string(),
+            ws: None,
+        });
+
+        if client_config.active_env.is_none() {
+            client_config.active_env = client_config.envs.first().map(|env| env.alias.clone());
```

Comment:
nit: I would just set this using a constant.

```suggestion
            client_config.active_env = Some("testnet".to_string())
```

Reply:
Added a proper function for local and testnet.